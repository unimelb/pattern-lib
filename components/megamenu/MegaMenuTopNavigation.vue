<template>
  <nav>
    <ul
      class="mega-top">
      <li
        v-for="(item, index) in items"
        :key="index">
        <a
          :href="item.href"
          class="menu-top__item">
          {{ item.title }}
        </a>
      </li>
    </ul>
  </nav>
</template>

<script>
export default {
  name: 'MegaMenuTopNavigation',
  props: {
    items: {
      type: Array,
      default: () => [],
    },
  },
};
</script>
