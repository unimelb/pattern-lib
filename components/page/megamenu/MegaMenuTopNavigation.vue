<template>
  <nav>
    <ul
      class="mega-top"
      role="menu">
      <li
        v-for="(item, index) in items"
        :key="index"
      >
        <a
          :href="item.href"
          role="menuitem"
          class="menu-top__item">
          {{ item.title }}
        </a>
      </li>
    </ul>
  </nav>
</template>

<script>
export default {
  name: 'MegaMenuTopNavigation',
  props: {
    items: {
      type: Array,
      default: () => [],
    },
  },
};
</script>
