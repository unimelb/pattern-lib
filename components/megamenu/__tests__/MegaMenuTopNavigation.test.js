import { shallow } from 'vue-test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import MegaMenuTopNavigation from '../MegaMenuTopNavigation.vue';

expect.extend(toHaveNoViolations);

describe('MegaMenuTitle', () => {
  it('should match snapshot', () => {
    const result = shallow(MegaMenuTopNavigation).element;
    expect(result).toMatchSnapshot();
  });

  it('should have default props and correct types items', () => {
    const wrapper = shallow(MegaMenuTopNavigation);
    const { items } = wrapper.vm.$options.props;

    expect(items.type).toBe(Array);
    expect(wrapper.props().items).toEqual([]);
  });

  it('should render items from prop', () => {
    const wrapper = shallow(MegaMenuTopNavigation, {
      propsData: {
        items: [
          {
            title: 'test',
            href: '/',
          },
        ],
      },
    });
    expect(wrapper.props().items.length).toBe(1);
    expect(wrapper.find('.menu-top__item').text()).toBe('test');
  });

  it('Component throws no accessibility violations', (done) => {
    const html = shallow(MegaMenuTopNavigation).html();
    // pass anything that outputs html to axe
    return axe(html).then((response) => {
      expect(response).toHaveNoViolations();
      done();
    });
  });
});
