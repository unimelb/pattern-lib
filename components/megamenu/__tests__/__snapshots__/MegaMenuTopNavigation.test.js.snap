// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MegaMenuTitle should match snapshot 1`] = `
<nav
  aria-label="Secondary navigation"
>
  <ul
    class="mega-top"
  />
</nav>
`;
